src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -23,7 +23,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-class ControllerResolver implements ControllerResolverInterface
+class ControllerResolver extends ArgumentResolver implements ControllerResolverInterface
```

Comment:
How would you envision this as a trait? By extending I make sure the children of this class will keep working. By making a trait I would have to also add this trait in the argument resolver.

I've thought about decoration instead, but that would break any child class implementing `doGetArguments`, which I think is done in Drupal for example (I could be wrong).